app/models/run_file.rb

```diff
@@ -6,16 +6,24 @@ class RunFile < ActiveRecord::Base
 
   def self.for_file(file)
     if file.respond_to?(:read)
-      RunFile.for_text(file.read)
+      RunFile.for_text(file.read, File.binary?(file.tempfile))
```

Comment:
tempfile is where the actual file gets uploaded to so that I can check if it is a binary file since File.binary?() needs a file path to check.  I'll look and see if there's a way to avoid doing this.